sqlglot/dialects/snowflake.py

```diff
@@ -195,6 +197,17 @@ class Parser(parser.Parser):
             ),
         }
 
+        ALTER_PARSERS = {
+            **parser.Parser.ALTER_PARSERS,  # type: ignore
+            "UNSET": lambda self: self._parse_alter_table_set_tag(unset=True),
+            "SET": lambda self: self._parse_alter_table_set_tag(),
+        }
+
+        def _parse_alter_table_set_tag(self, unset: bool = False) -> exp.Expression:
+            self._match_text_seq("TAG")
+            parser = t.cast(t.Callable, self._parse_id_var if unset else self._parse_conjunction)
```

Comment:
The reason this cast was added is documented here: https://stackoverflow.com/questions/70831013/mypy-got-function-expected-callable-any